pkg/gateway/operations/putobject.go

```diff
@@ -144,18 +143,15 @@ func handleCopy(w http.ResponseWriter, req *http.Request, o *PathOperation, copy
 		_ = o.EncodeError(w, req, gatewayErrors.Codes.ToAPIErr(gatewayErrors.ErrInvalidCopySource))
 		return
 	}
+	// check if src and dst are in the same repository and branch
 	var ent *catalog.DBEntry
-	// check if src and dst are in the same repository
-	if strings.EqualFold(o.Repository.Name, p.Repo) {
+	if o.Repository.Name == p.Repo && o.Reference == p.Reference {
```

Comment:
Stupid question, which was not introduced in this PR: Where do we validate that the reference is a branch and not a commit/tag?

Reply:
Gateway Put handle check if branch/reference exists before branching out to "copy"